File: src/main/java/travelator/camp/CampSites.java
package travelator.camp;

import static java.util.stream.Collectors.toUnmodifiableSet;

import java.util.Set;

public class CampSites {
    public static Set<CampSite> sitesInRegion(
        Set<CampSite> sites,
        String countryISO,
        String region
    ) {
        return sites.stream()
            .filter( campSite ->
                campSite.getCountryCode().equals(countryISO) &&
                    campSite.getRegion().equalsIgnoreCase(region)
            )
            .collect(toUnmodifiableSet());
    }
}

File: src/main/java/travelator/camp/CampSites.kt
package travelator.camp

fun Set<CampSite>.sitesInRegion(
    countryISO: String,
    region: String?
): Set<CampSite> = filter { site ->
    site.isIn(countryISO, region)
}.toSet()

fun CampSite.isIn(
    countryISO: String,
    region: String?
) = countryCode == countryISO &&
        region?.equals(this.region, ignoreCase = true) ?: true

